types/index.d.ts

```diff
@@ -340,20 +352,59 @@ export interface CookiesOptions {
 }
 
 /** [Documentation](https://next-auth.js.org/configuration/events) */
-export type EventType =
-  | "signIn"
-  | "signOut"
-  | "createUser"
-  | "updateUser"
-  | "linkAccount"
-  | "session"
-  | "error"
+export type EventCallback<MessageType = any> = (
+  message: MessageType
+) => Promise<void>
 
-/** [Documentation](https://next-auth.js.org/configuration/events) */
-export type EventCallback = (message: any) => Promise<void>
+export interface SessionEventMessage {
+  session: Session & Record<string, any>
+  jwt?: JWT
+}
```

Comment:
This `interface` doesn't seem to be used; if needed, could you let us know the use-case for it? 🧐

Reply:
Yup - my bad, missed that clean-up from an earlier draft.